File: api/models/movie.go
package models

import (
	"errors"
	"github.com/getsentry/sentry-go"
	"github.com/jinzhu/gorm"
)

type Movie struct {
	gorm.Model

	Title     string   `json:"title" gorm:"unique,not null"`
	Year      int      `json:"year"`
	AgeRating string   `json:"age_rating"`
	Duration  int      `json:"duration"`
	Genre     string   `json:"genre"`
	Country   string   `json:"country"`
	Synopsis  *string  `json:"synopsis"`
	Trailer   *string  `json:"trailer"`
	ImdbURL   *string  `json:"imdb_url"`
	Rating    *float64 `json:"rating"`
	Poster    *string  `json:"poster"`
}

func SaveMovie(movie *Movie) error {
	db := db.Where(Movie{Title: movie.Title}).FirstOrCreate(&movie)
	if db.Error != nil {
		sentry.CaptureException(db.Error)
		return errors.New("failed to create a movie. " + db.Error.Error())
	}

	if movie.ID <= 0 {
		sentry.CaptureException(errors.New("failed to create a movie, connection error"))
		return errors.New("failed to create a movie, connection error")
	}

	return nil
}

File: crawler/data/Movie.py
from dataclasses import dataclass


@dataclass
class Movie:
    title: str
    original_title: str
    year: int
    age_rating: str
    duration: int
    genre: str
    country: str
    version: str
    format: [None, str]
    synopsis: [None, str]
    trailer_url: [None, str]
    imdb_url: [None, str]
    rating: [None, float]

    def to_json(self):
        return {
            'title': self.title,
            'original_title': self.original_title,
            'year': self.year,
            'age_rating': self.age_rating,
            'duration': self.duration,
            'genre': self.genre,
            'country': self.country,
            'synopsis': self.synopsis,
            'trailer_url': self.trailer_url,
            'imdb_url': self.imdb_url,
            'rating': self.rating
        }



